newIDE/app/src/MainFrame/index.js

```diff
@@ -1715,8 +1663,9 @@ const MainFrame = (props: Props) => {
     return resourceSourceDialog.chooseResources(currentProject, multiSelection);
   };
 
-  const updateToolbar = (_state = state) => {
-    const editorTab = getCurrentTab(_state.editorTabs);
+  const updateToolbar = stateRef => {
```

Comment:
Not a big fan of "stateRef". `ref` has a meaning in React that I would rather not mix.
You might want to hint that this is a "reference" in JavaScript, but in JS all objects and arrays are reference, so this should not appear in the code.

Prefer `newState`. Also add the typing :)